documentation/design/proposals/authn_authz_for_1.0.adoc

```diff
@@ -0,0 +1,329 @@
+# Proposal - Authentication and Authorization for 1.0
+
+## Overview
+
+1.0 will introduce a new Access Control Service.  This will replace the existing authentication services.
+It will have responsibility for identity management and authorization for the messaging use-case.
+
+For identity, it will have provision for:
+
+* leverage an external identity provider (such as a LDAP provider),
+* leverage an external AMQP SASL authentication endpoint (implementing the `authServicePlugin` interface),
+* provide an internal identity provider (where users are represented by a kubernetes custom resources),
+* utilize client auth (in collabration with the `MessagingEndpoint` and the Qpid Dispatch Router),
+* utilise service account tokens for messaging authentication purposes, and
+* provide an anonymous identity provider (SASL ANONYMOUS)
+
+For authorization, authorization rules sourced from custom kubernetes resources will allow individual users, or
+groups of users to be permissioned for sending to or receiving from address(es) within the namespace.
+
+The Service Admin will be able to selectively enable/disable identity providers by configuring the Access Control Service
+resource. This will allow EnMasse to conform to the identity management rules of a particular organisation.  For instance, it will be possible for the Access Control Service to be configured so that only the external LDAP is available, or that
+service account and the external LDAP are available.
+
+Each Messaging Tenant will be associated with exactly one Access Control Service. This can be done locally by a Tenant
+Admin, or alternatively the Service Admin will be able to configure one or more Access Control Services centrally. Selectors will automatically associate an Access Control Service with the Tenants.
```

Comment:
By automatically associating, how would this work with overlapping selectors? Should it be possible to explicitly configure the default AccessControlService in a `MessagingPlan`?

Reply:
This comment was made against the old design, I don't think it applies to the new.